### PR TITLE
feat: implement rewrite path prefix filter for HTTPRoute in traditional router mode

### PR DESCRIPTION
**Pull request by czeslavo**
**What this PR does / why we need it**:

Implements `HTTPRoute`'s `URLRewrite` filter with `path` type `ReplacePrefixMatch` for the traditional router. Enables `SupportHTTPRoutePathRewrite` feature in Gateway API conformance tests.

Support for `expressions` router will be implemented in a separate PR.

**Which issue this PR fixes**:

Part of https://github.com/Kong/kubernetes-ingress-controller/issues/3686.

**PR Readiness Checklist**:

Complete these before marking the PR as `ready to review`:

- [x] the `CHANGELOG.md` release notes have been updated to reflect any significant (and particularly user-facing) changes introduced by this PR

### CODE REVIEW

File: .github/workflows/_integration_tests.yaml [hunk 1]
```diff
@@ -122,10 +122,12 @@ jobs:
           # Set enterprise to 'true' to enable isolated integration test cases requiring enterprise features.
           - name: isolated-dbless
             test: isolated.dbless
+            router-flavor: traditional
```
Comment (czeslavo) [inline .github/workflows/_integration_tests.yaml:125]: If we do not set it here, it falls back to `expressions` defined in `TEST_KONG_ROUTER_FLAVOR` env above.

File: internal/dataplane/translator/subtranslator/httproute.go [hunk 2]
```diff
@@ -544,43 +617,130 @@ func kongHeaderFormatter(header gatewayapi.HTTPHeader) string {
...
+
+	// If an empty path is provided, we need to make sure that the path will always start with a slash.
+	if path == "" {
+		path = "/"
+	}
+
+	// If the ReplacePrefixMatch is provided, we need to use it as the replacement URI.
+	if replacePrefixMatch := filter.Path.ReplacePrefixMatch; replacePrefixMatch != nil {
+		// Trim the trailing slash from the ReplacePrefixMatch to avoid double slashes in the final URI.
+		replaceURI = strings.TrimSuffix(*replacePrefixMatch, "/")
```
Comment (randmonkey) [inline internal/dataplane/translator/subtranslator/httproute.go:692]: How about extracting the logic of calculating replace URI to another function?

File: internal/dataplane/translator/subtranslator/httproute.go [hunk 3]
```diff
@@ -356,40 +358,57 @@ func mustMarshalJSON[T any](val T) string {
 // The plugins can be set in two different ways:
 // - Direct conversion from the respective HTTPRouteFilter.
 // - ExtensionRef to plugins annotation from the ExtensionRef filter.
-func SetRoutePlugins(route *kongstate.Route, filters []gatewayapi.HTTPRouteFilter, path string, tags []*string) error {
-	plugins, pluginAnnotation, err := generatePluginsFromHTTPRouteFilters(filters, path, tags)
+func SetRoutePlugins(route *kongstate.Route, filters []gatewayapi.HTTPRouteFilter, path string, tags []*string, expressionsRouterEnabled bool) error {
```
[the remark this answers is not in the document]
Comment (czeslavo) [reply internal/dataplane/translator/subtranslator/httproute.go]: Added comments: https://github.com/Kong/kubernetes-ingress-controller/pull/5895/commits/2334da27ca1849d4b4809b6ce8554a741ee35c5f

File: internal/dataplane/translator/subtranslator/httproute.go (hunk 2 above)
Comment (czeslavo) [reply internal/dataplane/translator/subtranslator/httproute.go]: Split `generateRequestTransformerForURLRewritePrefixMatch` to a function responsible for generating the request transformer's URI replace string, and another one for generating a Kong Route modifier: https://github.com/Kong/kubernetes-ingress-controller/pull/5895/commits/7c6e3a08e39168dcf119fc948ba199c9dde41f95

File: examples/gateway-httproute-rewrite-path.yaml [hunk 4]
```diff
@@ -100,3 +100,25 @@ spec:
...
+    - matches:
+        - path:
+            type: PathPrefix
+            value: /old-prefix
+      filters:
+        - type: URLRewrite
+          urlRewrite:
+            path:
+              type: ReplacePrefixMatch
+              replacePrefixMatch: "/echo"
```
Comment (programmer04) [inline examples/gateway-httproute-rewrite-path.yaml:121]:
```suggestion
              replacePrefixMatch: /echo
```
to be consistent with other paths in this yaml that are not quoted. I suppose this quote is meaningless here, isn't it?

File: internal/dataplane/translator/testdata/golden/httproute-url-rewrite-path-prefix/in.yaml [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    - matches:
+        - path:
+            type: PathPrefix
+            value: /prefix
+      filters:
+        - type: URLRewrite
+          urlRewrite:
+            path:
+              type: ReplacePrefixMatch
+              replacePrefixMatch: "/new-prefix"
```
Comment (programmer04) [inline internal/dataplane/translator/testdata/golden/httproute-url-rewrite-path-prefix/in.yaml:52]:
```suggestion
              replacePrefixMatch: /new-prefix
```
to be consistent with other paths in this yaml that are not quoted. I suppose this quote is meaningless here, isn't it?

**Review by randmonkey — APPROVED** [review]

**Review by programmer04 — APPROVED** [review]

**Pull request merged**